lab-integration/src/modules/tools/JsonFileTestsManager.ts

```diff
@@ -0,0 +1,33 @@
+import { promises as fs } from 'fs';
+import path from 'path';
+import { testsResult } from '../types';
+
+export default class JsonFileTestsManager {
```

Comment:
Do you know if we need a class? Can we read the JSON file every second and then check for any updates?

What if we have a for loop in `main.ts` to read the JSON file every second and check a timestamp to confirm the JSON has the new data? We can have `let` to keep the timestamp from the previous run.

Comparing the array's length is not a reliable way to find out. Based on the JSON data structure, we can compare the test time and the script's last run. 

What do you think?

Reply:
I think the second part is a good ideia (not rely in array's length comparison). But I think tha we need keep the logic in the class. So for me we implement just logic suggested by you.